src/parser/mod.rs

```diff
@@ -12289,6 +12352,137 @@ impl<'a> Parser<'a> {
+                    Ok(n) => Some(n),
+                    Err(_) => None,
+                };
+                Ok(Some(ShowStatementIn {
+                    clause,
+                    parent_type: Some(ShowStatementInParentType::Database),
+                    parent_name,
+                }))
+            }
+        } else if self.parse_keyword(Keyword::SCHEMA) {
+            if self.peek_keywords(&[Keyword::STARTS, Keyword::WITH]) {
+                Ok(Some(ShowStatementIn {
+                    clause,
+                    parent_type: Some(ShowStatementInParentType::Schema),
+                    parent_name: None,
+                }))
+            } else {
+                let parent_name = match self.parse_object_name(false) {
+                    Ok(n) => Some(n),
+                    Err(_) => None,
+                };
+                Ok(Some(ShowStatementIn {
+                    clause,
+                    parent_type: Some(ShowStatementInParentType::Schema),
+                    parent_name,
+                }))
+            }
+        } else if self.parse_keyword(Keyword::ACCOUNT) {
+            let parent_name = match self.parse_object_name(false) {
+                Ok(n) => Some(n),
+                Err(_) => None,
+            };
+            Ok(Some(ShowStatementIn {
+                clause,
+                parent_type: Some(ShowStatementInParentType::Account),
+                parent_name,
+            }))
+        } else if self.parse_keyword(Keyword::TABLE) {
+            let parent_name = match self.parse_object_name(false) {
+                Ok(n) => Some(n),
+                Err(_) => None,
+            };
+            Ok(Some(ShowStatementIn {
+                clause,
+                parent_type: Some(ShowStatementInParentType::Table),
+                parent_name,
+            }))
+        } else if self.parse_keyword(Keyword::VIEW) {
+            let parent_name = match self.parse_object_name(false) {
+                Ok(n) => Some(n),
+                Err(_) => None,
+            };
+            Ok(Some(ShowStatementIn {
+                clause,
+                parent_type: Some(ShowStatementInParentType::View),
+                parent_name,
+            }))
+        } else {
+            // Parsing MySQL style FROM tbl_name FROM db_name
+            // which is equivalent to FROM tbl_name.db_name
+            let mut parent_name = self.parse_object_name(false)?;
+            if self
+                .parse_one_of_keywords(&[Keyword::FROM, Keyword::IN])
+                .is_some()
+            {
+                parent_name.0.insert(0, self.parse_identifier(false)?);
+            }
+
+            Ok(Some(ShowStatementIn {
+                clause,
+                parent_type: None,
+                parent_name: Some(parent_name),
+            }))
+        }
+    }
+
+    fn parse_show_opt_starts_with(&mut self) -> Result<Option<Value>, ParserError> {
```

Comment:
```suggestion
    fn maybe_parse_show_starts_with(&mut self) -> Result<Option<Value>, ParserError> {
```
thinking similar here and for the others, the 'opt' in between reads somewhat like its part of the type being parsed, might be easily recognizable with the maybe prefix